cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFrameworkTest.java

```diff
@@ -155,8 +154,13 @@ public void onChange(ServiceDiscovered serviceDiscovered) {
 
     startLatch.await(5, TimeUnit.SECONDS);
 
+    // the framework.delete looks up namespace config through namespaceadmin add the meta there too.
+    NamespaceMeta systemMeta = new NamespaceMeta.Builder().setName(Id.Namespace.SYSTEM).build();
```

Comment:
also shouldn't location factory stuff happen automatically since you're using namespaceadmin?